### ISSUE TITLE
Think what to do with `iterable` interface

### ISSUE THREAD

**Issue by sobolevn** [dry-python/returns#503]
Should it be a part of specific interfaces or not?
@thepabloaguilar @astynax what do you think?

**Comment by thepabloaguilar** [comment]
I'm not sure, because will have to construct our own iterable types (e.g. list, tuple, set) to help the users.
On the other hand it'll be great, to do things like:
```python
# Do this:
flow(
    OurIterable(1, 2, 3, 4),
    map_(lambda number: number * 2)
) == OurIterable(2, 4, 6, 8)
# Instead of:
flow(
    Success([1, 2, 3, 4]),
    map_(lambda list_: list(map(lambda item: item * 2, list_)))
) == Success([2, 4, 6, 8])
```

**Comment by sobolevn** [comment]
No, that's not exactly the case. I was asking about this interface: https://github.com/dry-python/returns/blob/master/returns/interfaces/iterable.py Sorry, I need to provide links at the issue itself. 

My question is: should `IterableN` be a part of `*LikeN` / `*BasedN` types or not?

**Comment by thepabloaguilar** [comment]
> My question is: should `IterableN` be a part of `*LikeN` / `*BasedN` types or not?

Shouldn't, the `ResultBasedN` interface, for example, the user needs to implement **8 methods** if we add `IterableN` the user will have to implement **9 methods**, and maybe it won't use the method provided by `IterableN`!
This is my point of view, keep the interfaces (the specifics too) simple.

**Comment by sobolevn** [comment]
Closed by #528
